src/uct/cuda/cuda_ipc/cuda_ipc_md.c

```diff
@@ -28,6 +29,94 @@ static ucs_config_field_t uct_cuda_ipc_md_config_table[] = {
     {NULL}
 };
 
+typedef struct {
+    /* GPU Device number */
+    int     dev_num;
+    /* Cache of accessible devices (ucs_ternary_auto_value_t) */
+    uint8_t accessible[0];
+} uct_cuda_ipc_dev_cache_t;
+
+static UCS_F_ALWAYS_INLINE int uct_cuda_ipc_uuid_equals(CUuuid a, CUuuid b)
+{
+    int64_t *a64 = (int64_t *)a.bytes;
+    int64_t *b64 = (int64_t *)b.bytes;
+    return (a64[0] == b64[0]) && (a64[1] == b64[1]);
+}
+
+static UCS_F_ALWAYS_INLINE khint32_t uct_cuda_ipc_uuid_hash_func(CUuuid key)
+{
+    int64_t *i64 = (int64_t *)key.bytes;
+    return (khint32_t)(i64[0]);
```

Comment:
This would not change anything, because internally khash uses `khint_t` (which is `typedef khint32_t khint_t`) as a hash.
https://github.com/openucx/ucx/blob/master/src/ucs/datastruct/khash.h#L263
Btw, in the last version the impl uses all the bits to compute the hash, but still returns 32 bit integer:
```
static UCS_F_ALWAYS_INLINE khint32_t uct_cuda_ipc_uuid_hash_func(CUuuid key)
{
    int64_t *i64 = (int64_t *)key.bytes;
    return kh_int64_hash_func(i64[0] ^ i64[1]);
}
```